src/main/java/uk/gov/companieshouse/search/api/elasticsearch/AlphabeticalSearchRequests.java

```diff
@@ -0,0 +1,129 @@
+
+@Component
+public class AlphabeticalSearchRequests {
+
+    @Autowired
+    private RestClientService searchRestClient;
+
+    @Autowired
+    private EnvironmentReader environmentReader;
+
+    @Autowired
+    private AlphabeticalSearchQueries alphabeticalSearchQueries;
+
+    private static final String INDEX = "ALPHABETICAL_SEARCH_INDEX";
+    private static final String RESULTS_SIZE = "ALPHABETICAL_SEARCH_RESULT_MAX";
+    private static final String ORDERED_ALPHA_KEY_WITH_ID = "ordered_alpha_key_with_id";
+    private static final Logger LOG = LoggerFactory.getLogger(APPLICATION_NAME_SPACE);
+
+    public SearchHits getBestMatchResponse(String orderedAlphakey, String requestId) throws IOException {
+        SearchRequest searchRequestBestMatch = createBaseSearchRequest(requestId);
+        searchRequestBestMatch.source(bestMatchSourceBuilder(
+            alphabeticalSearchQueries.createOrderedAlphaKeySearchQuery(orderedAlphakey),
+            ORDERED_ALPHA_KEY_WITH_ID, SortOrder.ASC));
+
+        SearchResponse searchResponse = searchRestClient.searchRestClient(searchRequestBestMatch);
+        return searchResponse.getHits();
+    }
+
+    public SearchHits getStartsWithResponse(String orderedAlphakey, String requestId) throws IOException {
+        LOG.info("A hit was not found for: " + orderedAlphakey + ", falling back to prefix on alphakey");
+        SearchRequest searchRequestStartsWith = createBaseSearchRequest(requestId);
+
+        searchRequestStartsWith.source(bestMatchSourceBuilder(
+            alphabeticalSearchQueries.createOrderedAlphaKeyKeywordQuery(orderedAlphakey),
+            ORDERED_ALPHA_KEY_WITH_ID, SortOrder.ASC));
+
+        SearchResponse searchResponse = searchRestClient.searchRestClient(searchRequestStartsWith);
+        return searchResponse.getHits();
+    }
+
+    public SearchHits getCorporateNameStartsWithResponse(String orderedAlphakey,
+        String requestId) throws IOException {
+
+        LOG.info("A hit was not found for: " + orderedAlphakey + ", falling back to corporate name");
+        SearchRequest searchRequestCorporateName = createBaseSearchRequest(requestId);
+
+        // Consider using corporateName instead of orderedAlphakey
+        // Currently using same logic as python application
+        searchRequestCorporateName.source(bestMatchSourceBuilder(
+            alphabeticalSearchQueries.createStartsWithQuery(orderedAlphakey),
+            ORDERED_ALPHA_KEY_WITH_ID, SortOrder.ASC));
+
+        SearchResponse searchResponse = searchRestClient.searchRestClient(searchRequestCorporateName);
+        return searchResponse.getHits();
+    }
+
+    public SearchHits getAboveResultsResponse(String requestId,
+        String orderedAlphakeyWithId,
+        String topHitCompanyName) throws IOException {
+        LOG.info("Retrieving the alphabetically descending results for search elasticsearch: " + topHitCompanyName);
```

Comment:
Is this intended to be `...search elasticsearch:...`?

Reply:
Looks like a typo, will address